### DISCUSSION TITLE
Welcome text on startup?

### DISCUSSION THREAD

**Discussion by NikolaiRadke** [NikolaiRadke/Schimpfolino#3] [Ideas]
Should Schimpfolino send a greeting message when the battery is plugged in? It would be a way to personalize him.

**Comment by Birger-T** [comment]
First of all, Thank You - for this neat idea. At iobroker-forum guys implement it into their homeautomation with Alexa-Output ;-)

Why not only a pushbutton between battery an VCC? Should also work and could write swearwords to display, once or as long button is pressed?!

I made it with an ESP01 and extended wordlists; 
The adjectives  (in data1[]) should be stored without the last 'e' (for  'sch#ndlich', 'scheu*lich' etc.)
```
number = (random(0, addresses[0]));       // Select first word
    field = data1;                            // Pointer to first array
    get_swearword(number);                    // Read first word
    genus = random(0, 3);                     // Set word genus
    wordbuffer[chars++] = 'e';                // Adjective append an 'e'    ** new **
    if (genus != 0) {
      wordbuffer[chars++] = 'q' + genus;      // If male, write "r", if neutrum, write "s"
    }
    write_swearword(1);                       // Write first word in the first line
```
and to prevent the arduino compile error in the 'no_eeprom'-version
`char     *field;                                 // Pointer to one of the character arrays`
must be changed to
`const char     *field;                                 // Pointer to one of the character arrays`
and if
`uint16_t addresses[5] = {110, 110, 110, 110, 110}; // Wordlists addresses array - overwritten if EEPROM is present`
is changed to
```
uint16_t  addresses[5] = {sizeof(data1)/10,      // wordcount in array for first word (adjective)
                          sizeof(data2)/10,      // wordcount in array for second word part 1 noun-
                          sizeof(data3)/10,      // wordcount in array for second word part 2 female noun
                          sizeof(data4)/10,      // wordcount in array for second word part 2 male noun
                          sizeof(data5)/10};     // wordcount in array for second word part 2 neutrum noun

char      schimpfhello[] = {"Schimpfolino "};
```
you won't care about the wordlists len anymore!?
That's it for now, best regards /Birger

**Reply by NikolaiRadke** [reply]
Thanks for your Message!
I've never thought about the button between VCC and battery... There are no writing processes, so it would work without problems. Well, maybe next time :-)

I took a look at the iobroker forum - Cool idea! I will take a look every now and then.

You are right, the wordlist length is a stupid obstacle, so i adopted your code. And you are right with the redundant "e" but it won't reduce the size, it will only give the opportunity to find words with on more character. An adaption would mess the backward compatibility, so i will not touch it yet, but i keep it in mind.

Thanks for your help!

Nikolai
